### PR TITLE
User validation

### PR DESCRIPTION
**Pull request by Himalee**

### CODE REVIEW

File: lib/computer_player.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+require_relative "player"
+
+class ComputerPlayer < Player
```
Comment (pelensky) [inline lib/computer_player.rb]: Inheritance! Nice! 👍

File: lib/computer_player.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  attr_reader :player_input
+
+  def get_input
+    @display.present(Messages.new.player_prompt(@name))
```
Comment (pelensky) [inline lib/computer_player.rb]: 1) Where do we get `@name` from? 
2) You could argue this either way, but I'd probably prefer to see a different class prompt the player to get their input, and to display it once they've chosen. Would be keen to hear other's thoughts on this.

File: lib/computer_player.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    randomise
+    display_input
+  end
+
+  def randomise
+    options = ["rock", "paper", "scissors"]
+    @player_input = options.sample
```
Comment (pelensky) [inline lib/computer_player.rb]: 👍

File: lib/rules.rb [hunk 4]
```diff
@@ -1,20 +0,0 @@
-class Rules
```
Comment (pelensky) [inline lib/rules.rb]: What was the reason for moving this logic into the game class?

Comment (Himalee) [reply lib/rules.rb]: @pelensky I did have my logic in a 'Rules` class but I ended up having 3 if statements: one in `Rules` determining the shape that won, and two in game - which player had won and presenting the winning message. This way cut it down to two if statements. I could move that logic back to a `Rules` class - what do you think?

File: lib/computer_player.rb (hunk 2 above)
Comment (Himalee) [reply lib/computer_player.rb]: @pelensky I could have a new `GetInput` class?

Comment (sarahabimay) [reply lib/computer_player.rb]: Do you have a Display class already @Himalee ?

Comment (sarahabimay) [reply lib/computer_player.rb]: Maybe it could go in Display?

**Pull request merged**